sonar-scm-svn-plugin/src/main/java/org/sonar/plugins/scm/svn/SvnScmProvider.java

```diff
@@ -19,17 +19,33 @@
  */
 package org.sonar.plugins.scm.svn;
 
-
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
 import org.sonar.api.batch.scm.BlameCommand;
-import org.sonar.api.batch.scm.ScmProvider;
+import org.sonar.api.batch.scm.ScmBranchProvider;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.tmatesoft.svn.core.SVNException;
+import org.tmatesoft.svn.core.wc.SVNClientManager;
+import org.tmatesoft.svn.core.wc.SVNInfo;
+import org.tmatesoft.svn.core.wc.SVNLogClient;
+import org.tmatesoft.svn.core.wc.SVNWCClient;
 
-import java.io.File;
+import static org.sonar.plugins.scm.svn.SvnPlugin.newSvnClientManager;
 
-public class SvnScmProvider extends ScmProvider {
+public class SvnScmProvider extends ScmBranchProvider {
```

Comment:
As discussed, I think we should not introduce a new extension point, since it will prevent plugins to stay backward compatible. Adding the method branchChangedFiles to ScmProvider is better (and throw UnsupportedException by default)